### DISCUSSION TITLE
Utilize catalog.perseus.org data

### DISCUSSION THREAD

**Discussion by jacobwegner** [scaife-viewer/scaife-viewer.org#27] [Ideas]
_Notes (from ![](https://github.trello.services/images/mini-trello-icon.png) [Integrate catalog.perseus.org](https://trello.com/c/W0QLR6FO))_:

**Author information**

This seems like we could include information in a widget:

- https://github.com/PerseusDL/catalog_data/blob/master/mads/PrimaryAuthors/E/Epictetus/author.560.1.mads.xml
- Access via catalog.perseus.org, g.g. https://catalog.perseus.org/catalog/urn:cite:perseus:author.560

**Works in pipeline / not yet in TEI-XML / viewable on a Scaife site**

E.g. for a given URN, can we fall back to a viewable version of that URN?  If we don't have it on scaife.perseus.org, can we view it on http://www.perseus.tufts.edu/hopper?

If we don't even yet have it in the hopper, can we show the user a useful page on catalog.perseus.org?

Might want to take a closer look at https://github.com/PerseusDL/perseus_catalog too.

**Comment by jacobwegner** [comment]
Re: 
> Works in pipeline / not yet in TEI-XML / viewable on a Scaife site

I've spun up a small service that is the first iteration of resolving URNs.

I've hardcoded Iliad and Odyssey to resolve to beyond-translation.perseus.org:

```
https://catalog-api-dev.scaife.eldarion.com/urn:cts:greekLit:tlg0012.tlg001.perseus-grc2:22.481/canonical-url/
-->
https://beyond-translation.perseus.org/reader/urn:cts:greekLit:tlg0012.tlg001.perseus-grc2:22.481
```

Right now the `canonical-url` is hardcoded into the LSJ annotations; in the future, we might do something fancier within the DictionaryEntry widget to resolve / lookup using another method.

**Comment by jacobwegner** [comment]
On our dev call earlier this week, someone mentioned that catalog.perseus.org had a concept of canonical record URLs. 

Turns out the `data.perseus.org` service (which if it isn't the same code base as catalog.perseus.org seems similar) has a pretty useful endpoint that I'm now using in my redirect chain.  I hope to circle back to my service next week, but I made an update to the resolver service that utilizes that existing endpoint.


Looking at the `μῆνις` entry in LSJ:

https://beyond-translation.perseus.org/reader/urn:cts:greekLit:tlg0012.tlg001.perseus-grc2:1.1?entryUrn=urn%3Acite2%3Ascafife-viewer%3Adictionary-entries.atlas_v1%3Alsj-67481

<img width="1087" alt="image" src="https://user-images.githubusercontent.com/629062/169163030-53ea16c3-72c9-4fbb-8358-4f54dbae552d.png">

The first 6 refs get resolved as follows:


| Ref                                                          | URN                                               | Canonical URL                                                |
| ------------------------------------------------------------ | ------------------------------------------------- | ------------------------------------------------------------ |
| [Pl. *R.* 390e ](https://catalog-api-dev.scaife.eldarion.com/urn:cts:greekLit:tlg0059.tlg030.perseus-grc1:390e/canonical-url/) | urn:cts:greekLit:tlg0059.tlg030.perseus-grc1:390e | http://www.perseus.tufts.edu/hopper/text?doc=urn:cts:greekLit:tlg0059.tlg030.perseus-grc1:390e |
| [Ael. *Fr.* 80 ](https://catalog-api-dev.scaife.eldarion.com/urn:cts:greekLit:tlg0545.tlg004:80/canonical-url/) | urn:cts:greekLit:tlg0545.tlg004:80                | https://catalog.perseus.org/catalog/urn:cts:greekLit:tlg0545.tlg004 |
| [Them. *Or.* 22.265d](https://catalog-api-dev.scaife.eldarion.com/urn:cts:greekLit:tlg2001.tlg022:265d/canonical-url/) | urn:cts:greekLit:tlg2001.tlg022:265d              | https://catalog.perseus.org/catalog/urn:cts:greekLit:tlg2001.tlg022 |
| [Jul. *Or.*2.50b](https://catalog-api-dev.scaife.eldarion.com/urn:cts:greekLit:tlg2003.tlg001:2.50b/canonical-url/) | urn:cts:greekLit:tlg2003.tlg001:2.50b             | http://www.perseus.tufts.edu/hopper/invalidquery.jsp?doc=Perseus:text:2008.01.0657:orgpage=2 |
| [Il. 5.34](https://catalog-api-dev.scaife.eldarion.com/urn:cts:greekLit:tlg0012.tlg001.perseus-grc2:5.34/canonical-url/) | urn:cts:greekLit:tlg0012.tlg001.perseus-grc2:5.34 | https://beyond-translation.perseus.org/reader/urn:cts:greekLit:tlg0012.tlg001.perseus-grc2:5.34 |
| [A. *Ag.* 701](https://catalog-api-dev.scaife.eldarion.com/urn:cts:greekLit:tlg0085.tlg005.perseus-grc1:701/canonical-url/) | urn:cts:greekLit:tlg0085.tlg005.perseus-grc1:701  | http://www.perseus.tufts.edu/hopper/text?doc=urn:cts:greekLit:tlg0085.tlg005.perseus-grc1:701 |

**Reply by jacobwegner** [reply]
cc: @gregorycrane 

TLDR: We can resolve a lot more citations from LSJ entries now on Beyond Translation.

Mostly going to the catalog or Perseus 4, but I'll get scaife.perseus.org working with the resolver soon.

**Comment by jacobwegner** [comment]
I tried another pass at this today.  We will prefer scaife.perseus.org URNs, but in many cases, LSJ may use `grc1` where we now have `grc2` on scaife.perseus.org

As an example, given the following entry for [τίθημι](https://beyond-translation-dev.perseus.org/reader/urn:cts:greekLit:tlg0012.tlg001.perseus-grc2:1.2?mode=dictionary-entries&entryUrn=urn%3Acite2%3Ascafife-viewer%3Adictionary-entries.atlas_v1%3Alsj-103913)

The following citation:
>  Pl. R. 376e 

Has the URN `urn:cts:greekLit:tlg0059.tlg030.perseus-grc1:376e`.

If we try and resolve that to `grc2` on scaife.perseus.org, it won't work, due to milestone resolution:

https://scaife.perseus.org/reader/urn:cts:greekLit:tlg0059.tlg030.perseus-grc2:376e/

Other URNs may fare better:

`urn:cts:greekLit:tlg0059.tlg006.perseus-grc2:164c` -->
https://scaife.perseus.org/reader/urn:cts:greekLit:tlg0059.tlg006.perseus-grc2:164c/

It might be worth revisiting alternate chunking / milestones if we want to resolve these better to scaife.perseus.org.
